tests/transforms/test_split_non_commuting.py

```diff
@@ -59,7 +60,6 @@ def test_commuting_group_no_split(self, mocker):
             qml.Hadamard(0)
             qml.CNOT((0, 1))
             qml.expval(qml.PauliZ(0))
-            qml.expval(qml.PauliZ(0))
```

Comment:
Is it necessary to be removed ?

Reply:
Yes, because when having two equal measurements, `split_tape` will only execute one and copy the result. Thus the tape object won't be the same (which is what we are testing here).